xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/PropAddAction.java

```diff
@@ -80,13 +84,19 @@ public boolean action(XWikiContext context) throws XWikiException
         BaseClass bclass = doc.getXClass();
         bclass.setName(doc.getFullName());
         if (bclass.get(propName) != null) {
-            context.put("message", "action.addClassProperty.error.alreadyExists");
+            context.put("message", PROPERTY_ERROR_ALREADY_EXISTS_ERROR_MESSAGE);
             List<String> parameters = new ArrayList<String>();
             parameters.add(propName);
             context.put("messageParameters", parameters);
-            response.setStatus(HttpServletResponse.SC_BAD_REQUEST,
-                localizePlainOrKey("action.addClassProperty.error.alreadyExists", parameters.toArray()));
-            return true;
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            try {
+                String localizedMessage =
+                    localizePlainOrReturnKey(PROPERTY_ERROR_ALREADY_EXISTS_ERROR_MESSAGE, propName);
+                response.getWriter().write(localizedMessage);
```

Comment:
This sounds dangerous, when writing arbitrary content to the output, always also set the mime type to `text/plain` to avoid XSS attacks (no idea if this would be possible here, but better stay on the safe side).